src/textual/app.py

```diff
@@ -305,28 +310,33 @@ def bind(
             keys, action, description, show=show, key_display=key_display
         )
 
-    def run(self, loop: AbstractEventLoop | None = None) -> ReturnType | None:
+    def run(self) -> ReturnType | None:
         async def run_app() -> None:
             await self.process_messages()
 
-        if loop:
-            asyncio.set_event_loop(loop)
-        else:
-            try:
-                import uvloop
-            except ImportError:
-                pass
-            else:
-                asyncio.set_event_loop(uvloop.new_event_loop())
-
-        event_loop = asyncio.get_event_loop()
-        try:
+        if _ASYNCIO_GET_EVENT_LOOP_IS_DEPRECATED:
+            # N.B. This doesn't work with Python<3.10, as we end up with 2 event loops:
```

Comment:
took me a while to find something that make this function not crash for the 4 major versions of Python we support, with and without uvloop installed :sweat_smile: